### PR TITLE
chore(idFromtString): make error message friendlier - The error along is a bit cryptic, and if you're seeing it, the solution is somewhat buried in docs: https://developers.cloudflare.com/durable-objects/best-practices/access-durable-objects-from-a-worker/#parse-previously-created-ids-from-strings

### PR DESCRIPTION
**Pull request by thiskevinwang**

### CODE REVIEW

**Comment by thiskevinwang** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by DaniFoldi** [comment]
Hey 👋,

As someone who's run into this before, would it make sense to also mention `jurisdictions` in the error message? That also needs to match: https://developers.cloudflare.com/durable-objects/reference/data-location/#restrict-durable-objects-to-a-jurisdiction

**Comment by thiskevinwang** [comment]
@DaniFoldi I'm not able to test/repro jurisdictions related errors locally with `vitest-pool-workers`. I run into: `Error: Jurisdiction restrictions are not implemented in workerd.`.

Do the same requirements of having to first call `newUniqueId()` or `idFromName()` on the subnamespace apply?

I don't have strong opinions on including/omitting jurisdiction mention. It is also a _namespace_, so the current error message is still accurate w.r.t jurisdictions 😄

**Review by jasnell — APPROVED** [review]
I'm good with the change in general but I'm not sure it really provides much additional help.

**Comment by kentonv** [comment]
@DaniFoldi I don't think this particular error can be the result of using the wrong jurisdiction. If you use the wrong jurisdiction you'll get an ID that's valid for the namespace but points to a different object.

File: src/workerd/server/workerd-api.c++ [hunk 1]
```diff
@@ -605,7 +605,7 @@ public:
     // Verify that the computed mac matches the input.
     JSG_REQUIRE(memcmp(id + BASE_LENGTH, decoded.begin() + BASE_LENGTH,
                 decoded.size() - BASE_LENGTH) == 0,
-                TypeError, "Durable Object ID is not valid for this namespace.");
+                TypeError, "Durable Object ID is not valid for this namespace. Please ensure that the ID was previously created within the same namespace.");
```
[the remark this answers is not in the document]
Comment (thiskevinwang) [reply src/workerd/server/workerd-api.c++:608]: @kentonv Happy to rephrase it to anything else!

The main idea behind the additional error text is to provide remediation steps during testing, which is the angle I am/was coming from.

**For context** into what I was doing:
- I was using a "fake" Durable Object ID in a test, like so [this](https://github.com/thiskevinwang/durable-object-id-is-not-valid-for-this-namespace/blob/2199a4388a3d90d7c3f9827e57581a78db699454/test/index.test.ts#L16-L20), and hitting the error, which I was not aware how to resolve.
- I eventually stumbled my way through [the docs](https://developers.cloudflare.com/durable-objects/best-practices/access-durable-objects-from-a-worker/#parse-previously-created-ids-from-strings) and found out I had to call `newUniqueId()` or `idFromName()` in my tests first.
  - This additional trip through the docs is something I'm hoping the error message itself can help skip.

**Ideas** for rephrasing:
- Maybe the error text should mention those methods directly?
- Maybe the error text can link to docs? (This is a great DX, but it's an additional SDK-to-Web contract to maintain 😅 )

WDYT?
